lib/dialects/abstract/connection-manager.js

```diff
@@ -68,123 +67,123 @@ ConnectionManager.prototype.initPools = function () {
   var self = this
     , config = this.config;
 
-  if (config.replication) {
-    var reads = 0
-      , writes = 0;
+  if (!config.replication) {
+    this.pool = Pooling.Pool({
+      name: 'sequelize-connection',
+      create: function(callback) {
+        self.$connect(config).nodeify(function (err, connection) {
+          callback(err, connection); // For some reason this is needed, else generic-pool things err is a connection or some shit
+        });
+      },
+      destroy: function(connection) {
+        self.$disconnect(connection);
+      },
+      max: config.pool.max,
+      min: config.pool.min,
+      validate: config.pool.validate,
+      idleTimeoutMillis: config.pool.idle
+    });
+    return;
+  }
 
-    if (!Array.isArray(config.replication.read)) {
-      config.replication.read = [config.replication.read];
-    }
+  var reads = 0;
 
-    // Make sure we don't modify the existing config object (user might re-use it)
-    config.replication.write = _.extend({}, config.replication.write);
-    config.replication.read = config.replication.read.map(function (read) {
-      return _.extend({}, read);
-    });
+  if (!Array.isArray(config.replication.read)) {
+    config.replication.read = [config.replication.read];
+  }
+
+  // Make sure we don't modify the existing config object (user might re-use it)
+  config.replication.write = _.clone(config.replication.write);
+  config.replication.read = config.replication.read.map(function (read) {
```

Comment:
We might be able to replace this with [cloneDeep](https://lodash.com/docs#cloneDeep)